cwf/gateway/integ_tests/gx_reauth_test.go

```diff
@@ -93,10 +93,10 @@ func TestGxReAuthWithMidSessionPolicyRemoval(t *testing.T) {
 	recordsBySubID, err := tr.GetPolicyUsage()
 	assert.NoError(t, err)
 
-	record := recordsBySubID["IMSI"+imsi]["static-pass-all-raa1"]
+	record := recordsBySubID[prependIMSIPrefix(imsi)]["static-pass-all-raa1"]
 	assert.NotNil(t, record,
 		fmt.Sprintf("Policy usage record for imsi: %v rule: 'static-pass-all-raa1' does not exist", imsi))
-	assert.True(t, record.BytesTx > uint64(0), fmt.Sprintf("%s did not pass any data", record.RuleId))
+	assert.True(t, record.BytesTx+record.BytesRx > uint64(0), fmt.Sprintf("%s did not pass any data", record.RuleId))
```

Comment:
In all the tests IPERF only generates tx. There is no rx traffic being generated by the server (except for the generated by the execution of `checkIperfServerReachability`). So i would not change this in any test.

If we were doing this for both tx and rx we should do two `GenULTraffic` and then assert independently both traffics. I think adding the components on the assert in total form can overlook issues 

Btw, Unfortunately Iperf 3 removed the `bidirectional` flag that was on Iperf 2 that allowed to send and receive traffic at the same time.

Reply:
Hmm, that is odd. I think both rx and tx are both reported for the tests somehow.

Reply:
Recent log from CWF integ test run
```
Running TestGxReAuthWithMidSessionPoliciesRemoval...
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* TestRunner setup
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Adding Mock HSS service at 192.168.70.101:9204
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Adding Mock PCRF service at 192.168.70.101:9202
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Adding Mock OCS service at 192.168.70.101:9201
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Adding Pipelined service at 192.168.70.101:8443
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Adding Redis service at 192.168.70.101:6380
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Adding Directoryd service at 192.168.70.101:8443
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* Configuring 1 UE(s), PCRF instance: PCRF_REMOTE
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Added UE to Simulator, HSS_REMOTE, PCRF_REMOTE, and OCS_REMOTE:
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: 	IMSI: 000078153308272	Key: 20a901c66c25a27cfd489c8241c9d4c1	Opc: 6469f7d9fbb68149769a9a92e5d510fe	Seq: 924882308930
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Successfully configured UE(s)
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* Adding PCRF Usage Monitor for UE with IMSI: 000078153308272
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* Adding a Pass-All static rule: static-pass-all-raa1
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* Adding a Pass-All static rule: static-pass-all-raa2
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* Adding a Pass-All static rule: static-pass-all-raa3
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* Adding a base name mapping of base-raa1 -> [static-pass-all-raa3]
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* Adding PCRF Rule for UE with IMSI: 000078153308272 with ruleNames=[static-pass-all-raa1 static-pass-all-raa2], baseNames=[base-raa1]
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* Authenticating UE with IMSI: 000078153308272
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Finished Authenticating UE
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: ************************* Generating Traffic for UE with Req: imsi:"000078153308272" volume:<value:"450K" > 
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Record sid:"IMSI78153308272" rule_id:"static-pass-all-raa2" 
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Record sid:"IMSI78153308272" rule_id:"static-pass-all-raa3" 
[magma@10.240.0.9] out: [vagrant@127.0.0.1:2201] out: Record sid:"IMSI78153308272" rule_id:"static-pass-all-raa1" bytes_tx:424602 bytes_rx:68930 
```

Reply:
@uri200

Reply:
@themarwhal rx traffic is being generated in the form of tcp acks. I'm with the opinion to keep them both either as a separate asserts or total usage.

Reply:
Hey @themarwhal  as we discussed yesterday those acks are due to iperf not using TCP instead of UDP. 

If we are checking them as total and, for some reason, either uplink or downlink fails (for example due to a bad rule or QOS profile) you may be passing the test. To me, to prevent issues now and in the future, the best would be check them separately.